src/vtcomposite.cpp

```diff
@@ -692,104 +716,183 @@ struct LocalizeWorker : Napi::AsyncWorker
+                    std::vector<std::string> worldviews_to_create;
+
+                    // will be searching for the class with lowest index in class_key_precedence
+                    auto class_key_idx = static_cast<std::uint32_t>(class_key_precedence.size());
                     vtzero::property_value class_value;
-                    // accumulate final properties (except _mbx_worldview translation to worldview) here
-                    std::vector<std::pair<std::string, vtzero::property_value>> properties;
+
+                    auto language_key_idx = static_cast<std::uint32_t>(language_key_precedence.size());
+                    vtzero::property_value language_value;
+                    vtzero::property_value original_language_value;
+
+                    // collect final properties
+                    std::vector<std::pair<std::string, vtzero::property_value>> final_properties;
                     while (auto property = feature.next_property())
                     {
+                        // if already know we'll be skipping this feature, don't need to comb through its properties
+                        if (skip_feature)
+                        {
+                            continue;
+                        }
+
                         std::string property_key = property.key().to_string();
-                        if (!has_worldview_key && property_key == baton_data_->worldview_property)
+
+                        if (
+                            utils::startswith(property_key, baton_data_->worldview_property) ||
+                            utils::startswith(property_key, baton_data_->worldview_prefix + baton_data_->worldview_property))
                         {
-                            has_worldview_key = true;
-                            if (property.value().type() == vtzero::property_value_type::string_value)
+                            // skip feature only if the value of incompatible worldview key is not 'all'
+                            if (property_key == incompatible_worldview_key)
                             {
-                                worldviews_to_create = worldviews_for_feature(static_cast<std::string>(property.value().string_value()));
+                                if (property.value().type() == vtzero::property_value_type::string_value)
+                                {
+                                    if (property.value().string_value() != "all")
+                                    {
+                                        skip_feature = true;
+                                    }
+                                    // else do nothing - keep this feature but don't need to preserve this property.
+                                }
+                                else
+                                {
+                                    skip_feature = true;
+                                }
                             }
```

Comment:
Could you reduce this down to 
```
if (property_key == incompatible_worldview_key &&
                    (property.value().type() != vtzero::property_value_type::string_value ||
                    property.value().string_value() != "all")) {
	skip_feature = true;
}
```
I think that's logically equivlaent

Reply:
Yep that's logically equivalent, but harder for the human brain to comprehend. I prefer explicitly laying out the branches of the decision tree because past discussions on worldview showed that people find the concept of worldview and its implementation hard to grasp.

Reply:
I know I do! I find my suggested version easier to comprehend but it doesn't seem like that's true for everyone. 👍 to keeping as is.